### PR TITLE
feat: Add OpenJDK with CRaC patches

### PR DESCRIPTION
**Pull request by EyeCantCU**
Adds OpenJDK 17 and 21 with CRaC patches

### CODE REVIEW

**Review by xnox — DISMISSED** [review]
ship it

File: openjdk-crac-17.yaml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        # Test the jar
+        java -jar app.jar
+
+        # Test jlink
+        jlink --verbose --module-path "app.jar:$JAVA_HOME/jmods" \
+          --add-modules test.modules \
+          --output test-project-jre
+
+        # Test custom JRE
+        test-project-jre/bin/java -jar app.jar
```
Comment (xnox) [inline openjdk-crac-17.yaml:295]: Are there any symlistic java crac-criu processes that we can checkpoint restart?

Something like "sleep 10; print hello world", start it, checkpoint it, kill it, restart it, observe that it comes back and prints hello world? As that's the point of this build that one can do this.

File: openjdk-crac-17.yaml [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+      - libxext-dev
+      - libxi-dev
+      - libxrandr-dev
+      - libxrender-dev
+      - libxt-dev
+      - libxtst-dev
+      - openjdk-16
+      - openjdk-16-default-jvm
+      - zip
+      - crac-criu
```
Comment (xnox) [inline openjdk-crac-17.yaml:47]: needs wolfictl lint

File: openjdk-crac-17.yaml (hunk 1 above)
Comment (EyeCantCU) [reply openjdk-crac-17.yaml:294]: Wrote a small Java program that does this, but I can't get a package test working here as invoking CRIU for checkpoint/restore requires it to be running in a container with elevated privileges (or the NET_ADMIN cap set). We'll need to cover this in our image tests. Testing with a privileged wolfi-base container with OpenJDK CRaC and CRaC CRIU layered currently

Comment (xnox) [reply openjdk-crac-17.yaml:294]: fair enough!

**Pull request closed without merging**